src/main/flags/flags.ts

```diff
@@ -0,0 +1,74 @@
+export type ConfigFlags = {
+  enableMainlineDolphin: boolean;
+  enableReplayDatabase: boolean;
+};
+
+export enum RuntimeFlags {
+  ENABLE_MAINLINE_DOLPHIN = "--enable-mainline-dolphin",
+  ENABLE_REPLAY_DATABASE = "--enable-replay-database",
+}
+
+const DEVELOPMENT_CONFIG_FLAGS: ConfigFlags = {
+  enableMainlineDolphin: true,
+  enableReplayDatabase: true,
+};
+
+export function getConfigFlags(): ConfigFlags {
+  const isDevelopment = process.env.NODE_ENV === "development" || process.env.DEBUG_PROD === "true";
+  if (isDevelopment) {
+    return DEVELOPMENT_CONFIG_FLAGS;
+  }
+
+  const buildFlags = getBuildFlags();
+  const runtimeFlags = getRuntimeFlags();
+
+  return {
+    ...buildFlags,
+    ...runtimeFlags,
+  };
+}
+
+function getBuildFlags(): ConfigFlags {
+  return {
+    enableMainlineDolphin: parseBoolean(process.env.ENABLE_MAINLINE_DOLPHIN, false),
+    enableReplayDatabase: parseBoolean(process.env.ENABLE_REPLAY_DATABASE, false),
+  };
+}
+
+function getRuntimeFlags(): Partial<ConfigFlags> {
+  const overrides: Partial<ConfigFlags> = {};
+  const args = process.argv.slice(1);
+
+  let i = 0;
+  while (i < args.length) {
```

Comment:
We manually define the index outside the while loop so that in the future, if we want to handle flags which can accept an argument e.g. `--some-flag value` we can manually increment `i` by however many args we want to consume.
If we instead decide to use the format of `--some-flag=value` then we can remove this, but until then this keeps things flexible.